File: spec/system/breadcrumbs_spec.rb
require "rails_helper"

describe "breadcrumbs" do
  context "on the landing page" do
    it "links to NYU Libraries and the Library Catalog" do
      visit "/"

      within("ol.breadcrumb") do
        expect(page).to have_link("NYU Libraries", href: "https://library.nyu.edu")
        expect(page).to have_link("Library Catalog", href: "http://bobcat.library.nyu.edu/nyu")
        expect(page).to have_text("Archival Collections")
      end
    end
  end

  context "on brief results page" do
    it "links to Archival Collections and appends Search" do
      visit "/"

      fill_in "q", with: "foo"
      click_button "Search"

      within("ol.breadcrumb") do
        expect(page).to have_link("NYU Libraries", href: "https://library.nyu.edu")
        expect(page).to have_link("Library Catalog", href: "http://bobcat.library.nyu.edu/nyu")
        expect(page).to have_link("Archival Collections", href: "/")
        expect(page).to have_text("Search")
      end
    end
  end

  context "on repository/library pages" do
    it "links to repository/library" do
      visit "/fales"

      within("ol.breadcrumb") do
        expect(page).to have_link("NYU Libraries", href: "https://library.nyu.edu")
        expect(page).to have_link("Library Catalog", href: "http://bobcat.library.nyu.edu/nyu")
        expect(page).to have_link("Archival Collections", href: "/")
        expect(page).to have_link("The Fales Library & Special Collections", href: "/fales")
        expect(page).to have_text("Search")
      end
    end
  end
end

File: e2e/tests/breadcrumbs.spec.js
import {
  ATTRIBUTES,
  EXTERNAL_LINKS,
  LABELS,
  ROLES,
  ROUTES,
  SEARCH_TERMS,
} from './support/catalog-data.js';
import { expect, test } from '@playwright/test';
import { openHomepage, searchFromHome } from './support/catalog-interactions.js';

test.describe('Breadcrumbs', () => {
  test('landing page links to NYU Libraries and the Library Catalog', async ({ page }) => {
    await openHomepage(page);

    const breadcrumb = page.locator('ol.breadcrumb');
    await expect(breadcrumb.getByRole(ROLES.LINK, { name: LABELS.NYU_LIBRARIES })).toHaveAttribute(
      ATTRIBUTES.HREF,
      EXTERNAL_LINKS.NYU_LIBRARIES
    );
    await expect(
      breadcrumb.getByRole(ROLES.LINK, { name: LABELS.LIBRARY_CATALOG })
    ).toHaveAttribute(ATTRIBUTES.HREF, EXTERNAL_LINKS.LIBRARY_CATALOG);
    await expect(breadcrumb).toContainText(LABELS.ARCHIVAL_COLLECTIONS);
  });

  test('brief results page links to Archival Collections and appends Search', async ({ page }) => {
    await searchFromHome(page, SEARCH_TERMS.FOO);
    await expect(page).toHaveURL(new RegExp(`q=${SEARCH_TERMS.FOO}`, 'i'));

    const breadcrumb = page.locator('ol.breadcrumb');
    await expect(breadcrumb.getByRole(ROLES.LINK, { name: LABELS.NYU_LIBRARIES })).toHaveAttribute(
      ATTRIBUTES.HREF,
      EXTERNAL_LINKS.NYU_LIBRARIES
    );
    await expect(
      breadcrumb.getByRole(ROLES.LINK, { name: LABELS.LIBRARY_CATALOG })
    ).toHaveAttribute(ATTRIBUTES.HREF, EXTERNAL_LINKS.LIBRARY_CATALOG);
    await expect(
      breadcrumb.getByRole(ROLES.LINK, { name: LABELS.ARCHIVAL_COLLECTIONS })
    ).toHaveAttribute(ATTRIBUTES.HREF, ROUTES.HOME);
    await expect(breadcrumb).toContainText('Search');
  });

  test('repository page breadcrumb includes repository link', async ({ page }) => {
    await page.goto(ROUTES.FALES);

    const breadcrumb = page.locator('ol.breadcrumb');
    await expect(breadcrumb.getByRole(ROLES.LINK, { name: LABELS.NYU_LIBRARIES })).toHaveAttribute(
      ATTRIBUTES.HREF,
      EXTERNAL_LINKS.NYU_LIBRARIES
    );
    await expect(
      breadcrumb.getByRole(ROLES.LINK, { name: LABELS.LIBRARY_CATALOG })
    ).toHaveAttribute(ATTRIBUTES.HREF, EXTERNAL_LINKS.LIBRARY_CATALOG);
    await expect(
      breadcrumb.getByRole(ROLES.LINK, { name: LABELS.ARCHIVAL_COLLECTIONS })
    ).toHaveAttribute(ATTRIBUTES.HREF, ROUTES.HOME);
    await expect(breadcrumb.getByRole(ROLES.LINK, { name: LABELS.FALES_LIBRARY })).toHaveAttribute(
      ATTRIBUTES.HREF,
      ROUTES.FALES
    );
    await expect(breadcrumb).toContainText('Search');
  });
});
